### DISCUSSION TITLE
Cypress test responsive does not reflect the behaviour in real mobile device

### DISCUSSION THREAD

**Discussion by nudasilva** [cypress-io/cypress#22423] [Questions and Help]
Hello,

I currently use a Redmi 9, as device mobile, for my manual tests.
Note that the application, that i'm testing, has right tests responsive.

However, via Cypress, the same tests responsive is failing.

Here is the the viewport set at the beginning of the scenario, into Cypress - this well corresponds to the Redmi 9's one:

	beforeEach(() => {		
		cy.viewport(393, 873)
		...

Do I forget something in the script, please?

**Comment by jjhelguero** [comment]
To mimic web mobile testing, there are few configurations you'll need to set: `viewportWidth`, `viewportHeight`, and `userAgent`. The last being most important to receive the web mobile version of the app. 

In order to set these configurations, you'll have to pass them in via CLI. You can search online for valid userAgents for phone device and browser types

iOS:
`cypress open --config viewportWidth=400,viewportHeight=600,userAgent=\"Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) Mobile/14E304\"`

Android:
`cypress open --config viewportWidth=400,viewportHeight=600,userAgent=\"Mozilla/5.0 (Android 10; CPU SM-A205U) Mobile/14E304\"`

**Reply by nudasilva** [reply]
Thanks, but I'd used userAgent, as well, and I had the same result.

I've just tested with these userAgent values, and same issue.

In fact, it depends of the GUI, as regards responsive tests.
Cypress works for such a GUI, and for other.
